pkg/querier/queryrange/querysharding.go

```diff
@@ -200,18 +200,24 @@ func (ast *astMapperware) Do(ctx context.Context, r queryrangebase.Request) (que
 		return ast.next.Do(ctx, r)
 	}
 
-	v := ast.limits.TSDBShardingStrategy(tenants[0])
-	version, err := logql.ParseShardVersion(v)
-	if err != nil {
-		level.Warn(spLogger).Log(
-			"msg", "failed to parse shard version",
-			"fallback", version.String(),
-			"err", err.Error(),
-			"user", tenants[0],
-			"query", r.GetQuery(),
-		)
+	var strategy logql.ShardingStrategy
+
+	if conf.IndexType == types.TSDBType {
+		v := ast.limits.TSDBShardingStrategy(tenants[0])
+		version, err := logql.ParseShardVersion(v)
+		if err != nil {
+			level.Warn(spLogger).Log(
+				"msg", "failed to parse shard version",
+				"fallback", version.String(),
+				"err", err.Error(),
+				"user", tenants[0],
+				"query", r.GetQuery(),
+			)
+		}
+		strategy = version.Strategy(resolver, uint64(ast.limits.TSDBMaxBytesPerShard(tenants[0])))
+	} else {
```

Comment:
Should we be explicit about what index type supports NewPowerOfTwoStrategy (with an `else if`), and add a catch-all (`else`) that returns an error if we get an unexpected type? 

I'm worried this may cause a bug one day that's hard to track down if we add a new index type where this strategy doesn't work.